File: 백준/Bronze/2798. 블랙잭/블랙잭.py
#2798 블랙잭
# 브루트 포스 문제

num, blackjack = map(int, input().split(" "))
num_list = list(map(int, input().split(" ")))
new_list = []
for i in range(num):
    for j in range(i+1, num):
        for k in range(j+1, num):
            sum = num_list[i]+ num_list[j] + num_list[k]
            if sum > blackjack:
                continue
            else:
                new_list.append(sum)
print(max(new_list))

File: 백준/Bronze/2798. 블랙잭/블랙잭.kt
package com.example.kotlinstudy.bruteforce

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val (num, blackJack) = br.readLine().split(" ").map { it.toInt() }
    val blackJackArray = br.readLine().split(" ").map { it.toInt() }.toIntArray()
    var result = 0

    for (i in 0 until num - 2) {
        for (j in i + 1 until num - 1) {
            for (k in j + 1 until num) {
                val currentSum = blackJackArray[i] + blackJackArray[j] + blackJackArray[k]
                if (currentSum <= blackJack && currentSum > result) {
                    result = currentSum
                }
            }
        }
    }
    bw.write(result.toString())
    bw.flush()
    br.close()
    bw.close()
}